certbot-apache/certbot_apache/_internal/override_gentoo.py

```diff
@@ -1,36 +1,22 @@
 """ Distribution specific override class for Gentoo Linux """
-from typing import cast
-from typing import List
-
 import zope.interface
 
 from certbot import interfaces
 from certbot_apache._internal import apache_util
 from certbot_apache._internal import configurator
 from certbot_apache._internal import parser
+from certbot_apache._internal.configurator import _OsOptions
 
 
 @zope.interface.provider(interfaces.IPluginFactory)
 class GentooConfigurator(configurator.ApacheConfigurator):
     """Gentoo specific ApacheConfigurator override class"""
 
-    OS_DEFAULTS = dict(
+    OS_DEFAULTS = _OsOptions(
         server_root="/etc/apache2",
-        vhost_root="/etc/apache2/vhosts.d",
```

Comment:
Well, I am not that convinced it will improve the readability. On top of removing a lot of lines, I appreciate in my proposed approach that the `OsDefault` class contains the reference configuration that we can after override specifically, which is quite in line with the purpose of the override modules.

I also think that these parts, once properly configured in this PR, will not change much, so we can suppose this will not have a lot of influence in the future.

Also this is not about `restart_cmd_alt` which is not set, but also `ctl` or `logs_root` that are not set in `override_arch` for instance, so we will end up adding more duplication than before.

However independently from which decision we take here, I am surprised by how options are handled here. If we take again the exemple of `override_arch`, we see that `ctl` is not set. Does it mean that the function `_prepare_options` put `None` as the first entry for the arrays `version_cmd`, `restart_cmd` and `conftest_cmd`? I suppose this is not the case and a further logic circumvent that, but I fail to understand how this is working. Maybe we should invest some time to improve these handlers in the process?

Reply:
NB: I fixed the primary problem here about setting `vhost_root` for gentoo.

Reply:
> Well, I am not that convinced it will improve the readability. On top of removing a lot of lines, I appreciate in my proposed approach that the OsDefault class contains the reference configuration that we can after override specifically, which is quite in line with the purpose of the override modules.
>  
>  I also think that these parts, once properly configured in this PR, will not change much, so we can suppose this will not have a lot of influence in the future.

I still personally disagree as I think the value of `OS_DEFAULTS` in the base configurator class allows you to see the default values while keeping the definitions contained in one place in one file. In my opinion, the extra lines of code are worth it in this case for simplicity and readability. I'm not going to block this PR on this though so I wrote a [simple script](https://gist.github.com/bmw/ed480d838c21dc2878eed7aa374e84b5) to help me verify that no other values were mistakenly changed.

> Also this is not about restart_cmd_alt which is not set, but also ctl or logs_root that are not set in override_arch for instance, so we will end up adding more duplication than before.

I think you're mistaken and `ctl` and `logs_root` are set in every override class. Here are those values in `override_arch.py`: https://github.com/certbot/certbot/blob/e33090f2827558962984e0fa6cd16e653307f412/certbot-apache/certbot_apache/_internal/override_arch.py#L16-L17